rdrf/rdrf/admin.py

```diff
@@ -347,19 +418,43 @@ def groups(self, obj):
     groups.short_description = _("Allowed Groups")
 
 
+class RegistryRegistrationNotificationFilter(admin.SimpleListFilter):
+    title = _("Registry registration notifications")
+    parameter_name = 'registry_registration'
+
+    def lookups(self, request, model_admin):
+        return [(reg.code, reg.name) for reg in Registry.objects.all()]
+
+    def queryset(self, request, queryset):
+        if self.value():
+            registry = Registry.objects.get(code=self.value())
+            return queryset.filter(registry=registry, description__in=EventType.REGISTRATION_TYPES)
```

Comment:
You could do `registry__code=self.value()` directly in the filter and drop line 430.
I also don't think that restricting by `REGISTRATION_TYPES` is OK, as this is a general filter that isn't used just by the registration code, this PR is about.
If you wanted to achieve exactly what you have, ie. display only the registration related notifications, you would have to create a second filter that filter by `EventType`.
Or you could just drop the filtering by `description` at this stage and display all the notifications of the registry. In this case, you would also have to update the messages on the registration admin page "Current registration notifications belonging to the registry:" to "Current notifications belonging to the registry:".

Reply:
> I also don't think that restricting by REGISTRATION_TYPES is OK, as this is a general filter that isn't used just by the registration code, this PR is about.

I'm not sure what you mean by this? The filter is optional, and in the interface it says that it's filtering just registration notifications. It's a general filter, so it's not on by default and can be toggled by the user (when we merge in the filter interface), but it's also used by this PR because that's where the link goes in the `notifications_added.html` template:

```html
<a href="{% url "admin:rdrf_emailnotification_changelist" %}?registry_registration={{ registry.code }}">Link</a>
```

> If you wanted to achieve exactly what you have, ie. display only the registration related notifications, you would have to create a second filter that filter by EventType

In `EmailNotification`, the `description` field is where the type of notification (derived from `EventType`) is stored.

This filter is a custom filter that filters by both registry and registration type (through description). I don't think we could just use two filters, because there isn't an `or` operator in filters that would convey both the registration notification types (and possibly more in the future).

I'm a little confused by what you're asking for here. Perhaps we should discuss in person?